Pulled machine code out of Patches.py

mzxrules opened the pull request:
moved a significant amount of binary mips out of patches.py, and rewriting it in assembly in hacks.asm or new asm files. Bombchus in logic, gossip stone hint requirements, rainbow bridge requirements have been rewritten to be handled in asm. In the process of this rewrite, I removed the spidery hooks jumping into the 80400800 (03480800) address range.

TestRunnerSRL commented:
I wasn't able to compile the C code, I get the following error:
```c
rainbow_tunic.c: In function 'get_color':
rainbow_tunic.c:47:9: error: expected expression before '{' token
  return { r, g, b };
         ^
rainbow_tunic.c: In function 'update_color':
rainbow_tunic.c:54:7: error: 'cfg_zora_zora_tunic' undeclared (first use in this function); did you mean 'cfg_rainbow_zora_tunic'?
   || !cfg_zora_zora_tunic )
       ^~~~~~~~~~~~~~~~~~~
       cfg_rainbow_zora_tunic
rainbow_tunic.c:54:7: note: each undeclared identifier is reported only once for each function it appears in
```

Also there's a typo in `patches.py`
```python
    if world.bombchus_in_logic:
        rom.write_int32(rom.sym('RAINBOW_BRIDGE_CONDITION'), 1)
```
should be 
```python
    if world.bombchus_in_logic:
        rom.write_int32(rom.sym('BOMBCHUS_IN_LOGIC'), 1)
```

I also want to point out that in python, setting the parameter type hints doesn't really do anything. Python does not enforce that these hints are maintained, so this is only for various IDE to use some kind of intellisense and as documentation. I don't know how I feel about including extra imports just to maintain this.

mzxrules commented:
i removed the rainbow tunic stuff for now, as it isn't complete, and fixed the typo. 

as for the type hinting, I don't see a downside with making the code easier for an IDE to parse
